packages/server/validators/events.js

```diff
@@ -193,8 +193,44 @@ const eventIdValidator = [
     }),
 ];
 
+const checkInTimeValidator = [
+  param("eventId", "Invalid event ID")
+    .trim()
+    .blacklist("<>")
+    .isUUID()
+    .bail()
+    .withMessage("Not a UUID")
+    .exists({ checkFalsy: true })
+    .withMessage("Event Id cannot be null or empty")
+    .custom(async (eventId, { req }) => {
+      const userId = req.body.userId;
+      const event = await EventController.getEvent(eventId);
+      const isMember = await GuildController.isGuildMember(
+        event.guildId,
+        userId
+      );
+      if (!isMember) {
+        throw new Error("User is not a member of the guild.");
+      }
+
+      const currentTime = new Date();
+      const MINUTES_TO_MILLISECONDS = 60000;
+      const eventCheckInStartBound =
+        event.startDate.getTime() - 5 * MINUTES_TO_MILLISECONDS;
+      const eventCheckInEndBound =
+        event.startDate.getTime() + 15 * MINUTES_TO_MILLISECONDS;
+      const checkInStartTime = new Date(eventCheckInStartBound);
+      const checkInEndTime = new Date(eventCheckInEndBound);
+
+      if (currentTime < checkInStartTime || currentTime > checkInEndTime) {
+        throw new Error("Check-in window has closed.");
+      }
```

Comment:
We should have this error message be a bit more specific. It's weird if the frontend sees the error message "Check-in window has closed." before the event has even started. A different error message for before and after the check in time window would be better.

```js
if (currentTime < checkInStartTime) {
  throw new Error("This event has not opened for check-in yet.");
}
if (currentTime > checkInEndTime) {
  throw new Error("Check-in window has closed.");
}
```